cloudferrylib/os/network/neutron.py

```diff
@@ -109,9 +110,19 @@ def deploy(self, info):
         if self.config.migrate.keep_floatingip:
             self.upload_floatingips(deploy_info['networks'],
                                     deploy_info['floating_ips'])
+            if self.config.migrate.change_router_ips:
+                subnets_map = {subnet['id']: subnet
+                           for subnet in deploy_info['subnets']}
+                router_ips = self.extract_router_ips_as_floating_ips(
+                    subnets_map, deploy_info['routers'])
+                dst_router_ip_ids = self.upload_floatingips(
+                    deploy_info['networks'], router_ips)
         self.upload_routers(deploy_info['networks'],
                             deploy_info['subnets'],
                             deploy_info['routers'])
+        if self.config.migrate.clean_router_ips_stub and dst_router_ip_ids:
```

Comment:
I've designed another behavior.
Let's me describe it with more details:
Initial state: src cloud with router external ip 123.0.0.5 and FloatingIP 123.0.0.4
Migrate resources:

1. Move FloatingIP to dst. On dst we have FloatingIP 123.0.0.4
1. Create FloatingIP on dst as stub for router IP. On dst we have two FloatingIP [123.0.0.4, 123.0.0.5]. IP 123.0.0.5 exists only in OpenStack DB and not crush src network.
1. Create router on dst. **(here is the main idea)** As you see above, ips 123.0.0.4 and 123.0.0.5 already allocated, then OpenStack must allocate another ip for router (e.g. 123.0.0.6). PROFIT =)
1. FloatingIP 123.0.0.5 is not needed anymore. We use it on 1.3. step for not allow OpenStack create router with this ip. It can be released.

After resource migration we have
**src** router external ip 123.0.0.5 and FloatingIP 123.0.0.4
**dst** router external ip 123.0.0.6 and FloatingIP 123.0.0.4

Then we can migrate VMs in multiple steps.